src/utils/bytes.ts

```diff
@@ -11,425 +10,373 @@ if (isReactNative) {
-
-  if (typeof data === 'string') {
-    // If we're already a string return the string
-    // Strip leading 0x if it exists before returning
-    hexString = strip0x(data);
-  } else if (data instanceof BN) {
-    // If we're a BN object convert to string
-    // Return hex string 0 padded to even length, if length is 0 then set to 2
-    hexString = data.toString('hex', data.byteLength() * 2 || 2);
-  } else {
-    // We're an ArrayLike
-    // Coerce ArrayLike to Array
-    const dataArray: number[] = Array.from(data);
-
-    // Convert array of bytes to hex string
-    hexString = dataArray.map((byte) => byte.toString(16).padStart(2, '0')).join('');
-  }
+  // add 0x if it str isn't already prefixed
+  static prefix0x = (str: string): string => (isPrefixed(str) ? str : `0x${str}`);
 
-  // Return 0x prefixed hex string if specified
-  if (prefix) {
-    return `0x${hexString}`.toLowerCase();
-  }
+  // remove 0x prefix if it exists
+  static strip0x = (str: string): string => (isPrefixed(str) ? str.slice(2) : str);
 
-  // Else return plain hex string
-  return hexString.toLowerCase();
-}
+  static hexToBytes = hexToBytes;
 
-/**
- * Coerces BytesData into array of bytes
- * @param data - bytes data to coerce
- * @returns byte array
- */
-function arrayify(data: BytesData): number[] {
-  // If we're a BN object, convert to bytes array and return
-  if (data instanceof BN) {
-    // Else return bytes array
-    return data.toArray();
+  /**
+   * convert hex string to BigInt, prefixing with 0x if necessary
+   * @param {string} str
+   * @returns {bigint}
+   */
+  static hexToBigInt(str: string): bigint {
+    return BigInt(ByteUtils.prefix0x(str));
   }
 
-  // If we're already a byte array return array coerced data
-  if (typeof data !== 'string') {
-    return Array.from(data);
+  static u8ToBigInt(u8: Uint8Array): bigint {
+    // eslint-disable-next-line no-use-before-define
+    return ByteUtils.hexToBigInt(ByteUtils.hexlify(u8));
   }
 
-  // Remove leading 0x if exists
-  const dataFormatted = strip0x(data);
-
-  // Create empty array
-  const bytesArray: number[] = [];
-
-  // Loop through each byte and push to array
-  for (let i = 0; i < dataFormatted.length; i += 2) {
-    const number = parseInt(dataFormatted.substr(i, 2), 16);
-    if (Number.isNaN(number)) {
-      throw new Error('Invalid BytesData');
+  /**
+   * Coerces BytesData into hex string format
+   * @param data - bytes data to coerce
+   * @param prefix - prefix with 0x
+   * @returns hex string
+   */
+  static hexlify(data: BytesData, prefix = false): string {
+    let hexString = '';
+
+    if (typeof data === 'string') {
+      // If we're already a string return the string
+      // Strip leading 0x if it exists before returning
+      hexString = ByteUtils.strip0x(data);
+    } else if (typeof data === 'bigint' || typeof data === 'number') {
+      hexString = data.toString(16);
+      if (hexString.length % 2 === 1) {
+        hexString = `0${hexString}`;
```

Comment:
Does this cover the same scenarios as the original?

Reply:
Yes, but these specific lines of code here are new, they are covering bigints and numbers, which were not supported before.